### ISSUE TITLE
hi, how do the weights update?

### ISSUE THREAD

**Issue by qiumei1101** [jon-tow/surround-modulation-cnn#1]

**Comment by jon-tow** [comment]
Hi @qiumei1101 ! It's been a while since I've run this but you should be able to use it like any other `torch.nn.Module` and update the weights with a backwards call on a [cross-entropy loss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) term and an Adam step (as done in the paper).
![Screen Shot 2022-11-22 at 16 11 11](https://user-images.githubusercontent.com/41410219/203421929-ce794cca-6e79-44fa-a0dd-73a7a1e355a4.png)
